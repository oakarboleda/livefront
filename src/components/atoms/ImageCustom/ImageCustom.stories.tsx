import { Meta, StoryFn } from '@storybook/react'
import { ImageCustom } from './ImageCustom'
import React from 'react'

export default {
  title: 'Atoms/ImageCustom',
  component: ImageCustom,
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ['autodocs'],
  parameters: {
    // More on how to position stories at: https://storybook.js.org/docs/configure/story-layout
    layout: 'fullscreen',
  },
  args: {},
} satisfies Meta<typeof ImageCustom>

const Template: StoryFn<typeof ImageCustom> = (args) => <ImageCustom {...args} />

export const Primary = Template.bind({})
Primary.args = {
  // This is the default prop value for the component
  prop: 'default value',
}
