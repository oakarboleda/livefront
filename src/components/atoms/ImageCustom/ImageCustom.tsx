import React from 'react'

/**
 * Props for the ImageCustom component
 */
export interface imageCustomProps {
  /** An optional prop to customize the image text */
  prop?: string
}

/**
 * ImageCustom component
 *
 * A customizable image component that displays a text with an optional prop.
 *
 * @param {imageCustomProps} props - The props for the component.
 * @param {string} [props.prop='default value'] - An optional prop to customize the image text.
 * @returns {JSX.Element} The rendered image component.
 */
export function ImageCustom({ prop = 'default value' }: imageCustomProps): JSX.Element {
  return <div>ImageCustom {prop}</div>
}
