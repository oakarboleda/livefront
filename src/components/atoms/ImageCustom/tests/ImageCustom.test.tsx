import React from 'react';

import {ImageCustom} from '../ImageCustom';

describe('<ImageCustom />', () => {});
