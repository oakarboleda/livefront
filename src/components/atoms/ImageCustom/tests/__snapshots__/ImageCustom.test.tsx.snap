// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImageCustom /> matches snapshot 1`] = `
<DocumentFragment>
  <div>
    ImageCustom default value
  </div>
</DocumentFragment>
`;

exports[`<ImageCustom /> matches snapshot with custom prop 1`] = `
<DocumentFragment>
  <div>
    ImageCustom custom value
  </div>
</DocumentFragment>
`;
